### PR TITLE
Add Diagrams and [Add/Delete][Student/Component]Command Documentation to DG

### LINKED ISSUE #100: Provide Skeleton Sequence Diagram for DG
Provide a sequence diagram that shows overlapping logic flows for each command as well as setup phase of TutorLink, therefore allowing individual features to be depicted using concise and comprehensible `sd` diagrams.

### PR DESCRIPTION
**Pull request by RCPilot1604**
Fixes #100 

ArchitectureSequenceGrouped

### CODE REVIEW

File: docs/diagrams/AddStudentCommand.png
Comment (yeekian) [inline docs/diagrams/AddStudentCommand.png:1]: I like your use of reference frames for talking about separate commands

File: docs/diagrams/ArchitectureSequenceGrouped.png
Comment (yeekian) [inline docs/diagrams/ArchitectureSequenceGrouped.png:1]: Thanks for updating the architecture diagram!

**Review by yeekian — APPROVED** [review]
Good start for our DG, LGTM

File: docs/diagrams/ArchitectureSequenceGrouped.png
Comment (jinzihan2002) [inline docs/diagrams/ArchitectureSequenceGrouped.png:1]: At the beginning just after displayWelcomeMessage, should we include Storage loading ArrayLists from the data files, and initialising AppState with those ArrayLists?

**Review by jinzihan2002 — COMMENTED** [review]
Overall LGTM for now
Perhaps Storage loading data at the start could be added in the future

File: docs/diagrams/ArchitectureSequenceGrouped.png
Comment (RCPilot1604) [reply docs/diagrams/ArchitectureSequenceGrouped.png:1]: Yes

**Pull request merged**